ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -175,12 +180,19 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		if shouldUpdate := r.updateStatusToSuspendingIfNeeded(ctx, rayJobInstance); shouldUpdate {
 			break
 		}
+		if pastActiveDeadline(rayJobInstance) {
+			r.Log.Info("The RayJob has passed the activeDeadlineSeconds. Transition the status to `Complete`.", "StartTime", rayJobInstance.Status.StartTime, "ActiveDeadlineSeconds", *rayJobInstance.Spec.ActiveDeadlineSeconds)
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusComplete
+			break
+		}
 
-		// TODO (kevin85421): For light-weight mode, calculate the number of failed retries and transition
```

Comment:
We may consider implementing backoffLimit if ActiveDeadlineSeconds proves to be insufficient.